EthereumKit/Helper/SMP.swift

```diff
@@ -0,0 +1,2514 @@
+		//	z == Int.min as a special case.
+		if z == Int.min
+		{
+			self.init(sign: true, limbs: [Limb(Int.max) + 1])
+			return
+		}
+		else
+		{
+			self.init(sign: z < 0, limbs: [Limb(abs(z))])
+		}
+	}
+
+	/// Create an instance initialized to an unsigned integer value.
+	init(_ n: UInt)
+	{
+		self.init(limbs: [Limb(n)])
+	}
+
+	/// Create an instance initialized to a string value.
+	init(_ str: String)
+	{
+		var str = str
+		var sign = false
+		var base: Limbs = [1]
+		var limbs: Limbs = [0]
+
+		limbs.reserveCapacity(Int(Double(str.count) / log10(pow(2.0, 64.0))))
+
+		if str.hasPrefix("-")
+		{
+			str.remove(at: str.startIndex)
+			sign = str != "0"
+		}
+
+		for chunk in String(str.reversed()).split(19).map({ String($0.reversed()) })
+		{
+			if let num = Limb(String(chunk))
+			{
+				limbs.addProductOf(multiplier: base, multiplicand: num)
+				base = base.multiplyingBy([10_000_000_000_000_000_000])
+			}
+			else
+			{
+				fatalError("Error: String must only consist of Digits (0-9)")
+			}
+		}
+
+		self.init(sign: sign, limbs: limbs)
+	}
+
+	//	Requierd by protocol ExpressibleByFloatLiteral.
+	public init(floatLiteral value: Double)
+	{
+		self.init(sign: value < 0, limbs: [Limb(value)])
+	}
+
+	//	Required by protocol ExpressibleByIntegerLiteral.
+	public init(integerLiteral value: Int)
+	{
+		self.init(value)
+	}
+
+	// Required by protocol Numeric
+	public init?<T>(exactly source: T) where T : BinaryInteger
+	{
+		self.init(Int(source))
+	}
+
+	///	Creates an integer from the given floating-point value, rounding toward zero.
+	public init<T>(_ source: T) where T : BinaryFloatingPoint
+	{
+		self.init(Int(source))
+	}
+
+	///	Creates a new instance from the given integer.
+	public init<T>(_ source: T) where T : BinaryInteger
+	{
+		self.init(Int(source))
+	}
+
+	///	Creates a new instance with the representable value that’s closest to the given integer.
+	public init<T>(clamping source: T) where T : BinaryInteger
+	{
+		self.init(Int(source))
+	}
+
+	///	Creates an integer from the given floating-point value, if it can be represented
+	///	exactly.
+	public init?<T>(exactly source: T) where T : BinaryFloatingPoint
+	{
+		self.init(source)
+	}
+
+	///	Creates a new instance from the bit pattern of the given instance by sign-extending or
+	///	truncating to fit this type.
+	public init<T>(truncatingIfNeeded source: T) where T : BinaryInteger
+	{
+		self.init(source)
+	}
+
+	//
+	//
+	//	MARK: - CustomStringConvertible conformance
+	//	————————————————————————————————————————————————————————————————————————————————————————
+	//	||||||||        CustomStringConvertible conformance        |||||||||||||||||||||||||||||
+	//	————————————————————————————————————————————————————————————————————————————————————————
+	//
+	//
+	//
+
+	public var description: String
+	{
+		return (self.sign ? "-" : "").appending(self.limbs.decimalRepresentation)
+	}
+
+	public init(number: String, withBase base: Int)
+	{
```

Comment:
IMO: `number` parameter should be string/numberString😅 its a bit confusing. (no need to fix it)